datadog/resource_datadog_authn_mapping.go

```diff
@@ -0,0 +1,203 @@
+package datadog
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+
+	"github.com/DataDog/datadog-api-client-go/api/v2/datadog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/terraform-providers/terraform-provider-datadog/datadog/internal/utils"
+)
+
+func resourceDatadogAuthnMapping() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Provides a Datadog AuthN Mappings resource. This can be used to create and manage Datadog SAML Mappings.",
+		CreateContext: resourceDatadogAuthnMappingCreate,
+		ReadContext:   resourceDatadogAuthnMappingRead,
+		UpdateContext: resourceDatadogAuthnMappingUpdate,
+		DeleteContext: resourceDatadogAuthnMappingDelete,
+
+		Schema: map[string]*schema.Schema{
+			"key": {
+				Description: "Identity provider key.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"value": {
+				Description: "Identity provider value.",
+				Type:        schema.TypeString,
+				Required:    true,
+			},
+			"role": {
+				Description: "The role to assign for key:value mapping.",
```

Comment:
Would be good to clarify this is the ID of a role, not the full role resource.

"The ID of a role to attach to all users with the corresponding key and value"?